### ISSUE TITLE
hyper parameters of 'only_process_trajectory'

### ISSUE THREAD

**Issue by ChineseFootball10** [cocoon2wong/SocialCircle#10]
Hi @cocoon2wong 
      I've noticed in sdd dataset json file, 'only_process_trajectory' is set to 0. in EV^2-Net paper, i found it preprocess trajectory, bounding box and so many types. I would like to know where the arg 'only_process_trajectory' is defined, and if i didn't set --only_process_trajectory 0 what's the influence it will be. Thanks!
      Young.

**Comment by cocoon2wong** [comment]
Hi @ChineseFootball10,

This arg was designed to preprocess ego trajectories separately, without the inclusion of neighbors. In fact, it is not enabled and has been removed in later versions. It does not affect the kinds of trajectories that the network can predict.

**Comment by ChineseFootball10** [comment]
> Hi [@ChineseFootball10](https://github.com/ChineseFootball10),
> 
> This arg was designed to preprocess ego trajectories separately, without the inclusion of neighbors. In fact, it is not enabled and has been removed in later versions. It does not affect the kinds of trajectories that the network can predict.

Thank you so much. How about '--deterministic 0' in evscp json file?

**Comment by cocoon2wong** [comment]
@ChineseFootball10 

This arg is also useless now, considering our baselines are all generative models. It has also been removed in later versions, like that used in SocialCirclePlus repo.

You can use the global search or find references functions to look up the effect of these args in your code editor, since we have coded type hints for these parameters in our codes.

**Comment by ChineseFootball10** [comment]
> [@ChineseFootball10](https://github.com/ChineseFootball10)
> 
> This arg is also useless now, considering our baselines are all generative models. It has also been removed in later versions, like that used in SocialCirclePlus repo.
> 
> You can use the global search or find references functions to look up the effect of these args in your code editor, since we have coded type hints for these parameters in our codes.

Hi, @cocoon2wong 
Thank you so much. I would also want to know results of nba50k, since best ade was saved in 'best_epoch.txt', I found it was about 3.61 on nba dataset. It means ade of 0,5,9 steps.But in socialcircle paper, 10 frames results was 1.46. What was the problem?

<img width="230" alt="Image" src="https://github.com/user-attachments/assets/2d1d2d60-8999-4b34-be20-4119d6a9ce2f" />

<img width="245" alt="Image" src="https://github.com/user-attachments/assets/9cb534c9-405e-42f6-863f-56d01badefa1" />
I used python main.py --sc 'checkpoint_folder' to validate.

**Comment by ChineseFootball10** [comment]
> [@ChineseFootball10](https://github.com/ChineseFootball10)
> 
> This arg is also useless now, considering our baselines are all generative models. It has also been removed in later versions, like that used in SocialCirclePlus repo.
> 
> You can use the global search or find references functions to look up the effect of these args in your code editor, since we have coded type hints for these parameters in our codes.

Hi, @cocoon2wong 
      Sorry to disturb you. I'd like to know whether nba50k pretrained weights of socialcircle+ you provided at https://github.com/cocoon2wong/Project-Monandaeg/tree/SocialCirclePlus is available to reproduce results in paper? e.g. I used --sc instructure to load your pretrained weights on EVSCP model and got following results:

<img width="946" alt="Image" src="https://github.com/user-attachments/assets/4ea5d374-3382-41da-9003-bce39f5c8080" />

If changing fde(5.65) foot to meters, it is 1.72 meters which has gaps between results in paper. Same phenomenon occured at VSCP when I reproduced results on my device. Does this occurs on your device or I have something ignored? Thank you!
Young.

**Comment by cocoon2wong** [comment]
Hi @ChineseFootball10,

Could you please provide your test sets (which have been printed in the terminal outputs) ? 

Since the complete data in this dataset is much larger than the 50k egos we randomly sampled, the results will correspond to fluctuations when the train/test split is vastly different. The dataset split we used for training can be downloaded at https://github.com/cocoon2wong/Project-Luna/releases.

Here are my outputs:
```
>>> [Agent Manager]: 
    - Dataset name: NBA.
    - Dataset annotation type: coordinate.
    - Split name: nba50k.
    - Clips to test: ['11.30.2015.HOU.at.DET_316', '11.08.2015.DET.at.POR_51', '12.12.2015.LAL.at.HOU_365', '11.08.2015.LAL.at.NYK_202', '01.20.2016.PHI.at.ORL_151', '12.11.2015.CHA.at.MEM_331', '01.09.2016.MIA.at.UTA_25', '11.25.2015.ATL.at.MIN_476', '01.20.2016.PHI.at.ORL_304', '11.30.2015.HOU.at.DET_335', '...'].

>>> [SilverballersModel(0x159d376d0)]: 
    - Model name: model.
    - Prediction type: coordinate.
    - Prediction settings: 5 frames -> 10 frames, interval = 0.4 seconds.
    - Pre/post-process layers: [].
    - Network 0: EVSCPlusModel(0x158e1bc10).
    - Network 1: LinearSpeedHandlerModel(0x159198290).

...

>>> [Train Manager]: Test Results
    - ADE@2.0s(Metrics): 2.1646952629089355 (foot).
    - FDE@2.0s(Metrics): 2.8442063331604004 (foot).
    - ADE@4.0s(Metrics): 3.7434122562408447 (foot).
    - FDE@4.0s(Metrics): 4.490023612976074 (foot).
    - Average Inference Time: 109 ms.
    - Fastest Inference Time: 104 ms.
 
[INFO] `Train Manager` (SilverballersMKII): Test done. Split: nba50k, load: null, metrics: {'ADE@2.0s(Metrics)': '2.1646952629089355 (foot)', 'FDE@2.0s(Metrics)': '2.8442063331604004 (foot)', 'ADE@4.0s(Metrics)': '3.7434122562408447 (foot)', 'FDE@4.0s(Metrics)': '4.490023612976074 (foot)', 'Average Inference Time': '109 ms', 'Fastest Inference Time': '104 ms'}.
[INFO] `Train Manager` (SilverballersMKII): Test with subnetworks ['../Project-Monandaeg-SC+/evspcnba50k_adaptive', 'null'] done.
```

**Comment by ChineseFootball10** [comment]
> Hi [@ChineseFootball10](https://github.com/ChineseFootball10),
> 
> Could you please provide your test sets (which have been printed in the terminal outputs) ?
> 
> Since the complete data in this dataset is much larger than the 50k egos we randomly sampled, the results will correspond to fluctuations when the train/test split is vastly different. The dataset split we used for training can be downloaded at https://github.com/cocoon2wong/Project-Luna/releases.
> 
> Here are my outputs:
> 
> ```
> >>> [Agent Manager]: 
>     - Dataset name: NBA.
>     - Dataset annotation type: coordinate.
>     - Split name: nba50k.
>     - Clips to test: ['11.30.2015.HOU.at.DET_316', '11.08.2015.DET.at.POR_51', '12.12.2015.LAL.at.HOU_365', '11.08.2015.LAL.at.NYK_202', '01.20.2016.PHI.at.ORL_151', '12.11.2015.CHA.at.MEM_331', '01.09.2016.MIA.at.UTA_25', '11.25.2015.ATL.at.MIN_476', '01.20.2016.PHI.at.ORL_304', '11.30.2015.HOU.at.DET_335', '...'].
> 
> >>> [SilverballersModel(0x159d376d0)]: 
>     - Model name: model.
>     - Prediction type: coordinate.
>     - Prediction settings: 5 frames -> 10 frames, interval = 0.4 seconds.
>     - Pre/post-process layers: [].
>     - Network 0: EVSCPlusModel(0x158e1bc10).
>     - Network 1: LinearSpeedHandlerModel(0x159198290).
> 
> ...
> 
> >>> [Train Manager]: Test Results
>     - ADE@2.0s(Metrics): 2.1646952629089355 (foot).
>     - FDE@2.0s(Metrics): 2.8442063331604004 (foot).
>     - ADE@4.0s(Metrics): 3.7434122562408447 (foot).
>     - FDE@4.0s(Metrics): 4.490023612976074 (foot).
>     - Average Inference Time: 109 ms.
>     - Fastest Inference Time: 104 ms.
>  
> [INFO] `Train Manager` (SilverballersMKII): Test done. Split: nba50k, load: null, metrics: {'ADE@2.0s(Metrics)': '2.1646952629089355 (foot)', 'FDE@2.0s(Metrics)': '2.8442063331604004 (foot)', 'ADE@4.0s(Metrics)': '3.7434122562408447 (foot)', 'FDE@4.0s(Metrics)': '4.490023612976074 (foot)', 'Average Inference Time': '109 ms', 'Fastest Inference Time': '104 ms'}.
> [INFO] `Train Manager` (SilverballersMKII): Test with subnetworks ['../Project-Monandaeg-SC+/evspcnba50k_adaptive', 'null'] done.
> ```

Thank you for your reply! Here is my test clip which was printed on console platform.

<img width="947" alt="Image" src="https://github.com/user-attachments/assets/1d2a1743-3a7b-4a0b-894d-a4a05c9acd81" />

This test split was constructed as your README file.

<img width="640" alt="Image" src="https://github.com/user-attachments/assets/0f91870a-f629-489b-b983-105405da70b5" />

I manully put dataset_config and dataset_processed files from https://github.com/cocoon2wong/Project-Monandaeg/tree/main/Dataset. Does this same as https://github.com/cocoon2wong/Project-Luna/releases?

**Comment by cocoon2wong** [comment]
@ChineseFootball10 

Files of NBA set that from https://github.com/cocoon2wong/Project-Monandaeg/tree/main/Dataset and  https://github.com/cocoon2wong/Project-Luna/releases are exactly the same, which is the same split we use for training.

The sampled nba scenes will be different when they are manually constructed each time.

**Comment by cocoon2wong** [comment]
@ChineseFootball10 

Files of NBA set that from https://github.com/cocoon2wong/Project-Monandaeg/tree/main/Dataset and  https://github.com/cocoon2wong/Project-Luna/releases are exactly the same, which is the same split we use for training.  Our pretrained models may perform the same under this setting.

The sampled nba scenes will be different when they are manually constructed each time.

Have you changed other settings?

**Comment by ChineseFootball10** [comment]
> [@ChineseFootball10](https://github.com/ChineseFootball10)
> 
> Files of NBA set that from https://github.com/cocoon2wong/Project-Monandaeg/tree/main/Dataset and https://github.com/cocoon2wong/Project-Luna/releases are exactly the same, which is the same split we use for training. Our pretrained models may perform the same under this setting.
> 
> The sampled nba scenes will be different when they are manually constructed each time.
> 
> Have you changed other settings?

I haven't changed other settings since all hyperparameters of EVSCP were same as args.json file. Actually, I could load pretrained weights you provided by '--sc' instruction successfully. I checked size of nba trainset and testset printed on console platform. Train size on my device was 32273 and test size on my device was 12259. Does this same as yours?

<img width="716" alt="Image" src="https://github.com/user-attachments/assets/b6ac6944-5e57-4a02-ab73-da6529ccfe26" />

**Comment by cocoon2wong** [comment]
@ChineseFootball10 

It seems that we now have the same dataset settings. Do you get the same results as we do under this split? I think if the settings are exactly the same, the performance of these models should be the same as the results I provided above, and the same as the results in our paper as well.

**Comment by ChineseFootball10** [comment]
> [@ChineseFootball10](https://github.com/ChineseFootball10)
> 
> It seems that we now have the same dataset settings. Do you get the same results as we do under this split? I think if the settings are exactly the same, the performance of these models should be the same as the results I provided above, and the same as the results in our paper as well.

I have trained on my own device remaining all settings your provided fixed. E-V^2-Net-SC model got 1.25 ADE and 1.61 FDE. It's now clear that E-V^2-Net-SC is better than E-V^2-Net 1.26/1.64 which verify effectiveness of SocialCircle modeling but It's just that I haven't reproduced the results in the paper yet. I've tried to use '--restore' instructure to load checkpoint and continue trainng with much more little batch_size and learning_rate to find whether it could get better performance.

**Comment by ChineseFootball10** [comment]
Hi, @cocoon2wong 👍 
      I'd like to ask what's the role of val sets. In sdd split, val sets are not same as test sets

![Image](https://github.com/user-attachments/assets/8496b8fc-f296-4a82-8857-9cdff7ea5222)

![Image](https://github.com/user-attachments/assets/38c3cd2e-ffd7-492c-ae0d-99b39554fbf8)

But in train function, why https://github.com/cocoon2wong/Project-Qpid/blob/db144af30fda8b76e946756368ececd7039c62fc/training/__structure.py val_loader ds_val loaded test_sets?

<img width="425" alt="Image" src="https://github.com/user-attachments/assets/ed97f07e-dab8-47c1-92b8-10da5fdd302f" />

In 253 row, why not self.split_manager.val_sets?
I noticed in almost all previous work in pedestrain trajectory prediction, they only had train sets and test sets but they didn't have val sets. Are self.split_manager.train_sets corresponding their train sets and self.split_manager.test_sets corresponding their test set? If so, can we neglect self.split_manager.val_sets or can we add val set into train set to participate training as well (But if we do so, we must rerun all baseline models to verify their performance for training sets are larger they are possible to get better performance). 
ps. ETH-UCY adopt leave-one out strategy in social circle paper said. So train sets can be equal to test sets. But why sdd and nba sets didn't use val sets?
1) What's the use of self.split_manager.val_sets? 
2) Did you add val sets into train sets to train your models?
3) Another question is about anntype of sdd dataset. In ./dataset_original/codes/sdd/configs.py file, anntype is 'boundingbox', if it matters in args.json file anntype is 'coordinate'?
Thank you!
Young.

**Comment by cocoon2wong** [comment]
Hi @ChineseFootball10,

1. As you mentioned, most trajectory prediction papers directly use test sets as validation sets (i.e., those sets that should be tested during training). As a result, we still reserve locations in the code where the validation set should be used to facilitate future extensions and modifications. However, in the experiments, to keep the consistency of the comparison, the test set was tested during training and the data in the validation set was not used throughout the training or testing process. If you want to use validation sets to test during training, you can simply change this line (253 in your figure).
2. The direct use of the test set as the validation set also exists in other computer vision tasks. However, it should be noted that neither test sets nor validation sets should be used as part of the training set at all. We never use these validation sets as training sets.
3. The annotation type list in files in `dataset_configs` is their original annotations. The SDD is labeled in pixel as bounding boxes, which you can see in their original data files. The arg `--anntype` indicate how the code will process these original data and pass them to trajectory prediction models. You can see https://github.com/cocoon2wong/Project-Qpid/blob/main/dataset/__base/__picker.py.

**Comment by cocoon2wong** [comment]
@ChineseFootball10 

Do you have any other concerns? If not please close this issue or open some new ones if there are any. Thank you again for your questions!
